src/plugins/layout/layout.ts

```diff
@@ -230,10 +246,28 @@ export default class Layout implements IPlugin {
       };
       this.checkConsistency(actionInfo);
       if (this.allowMutation()) {
-        this.processMutations(summary, time);
+        const stateGenStartTime = getTimestamp();
+        this.processMutations(summary, mutationStartTime);
+        stateGenCount = (
+          summary.newNodes.length
+          + summary.movedNodes.length
+          + summary.updatedNodes.length
+          + summary.removedNodes.length
+        );
```

Comment:
do you think it would be useful at all to log each of these numbers rather than the aggregate? Just thinking about future debugging where it might be helpful to know at a glance we are getting pegged doing updates vs new nodes or something similar

Reply:
Hm, yeah, i think there is value in that. Right now we gen full state for any event (major perf hell, but nonetheless). However, such breakdown would give us an idea of how much perf we're wasting.

---
In reply to: [261791739](https://github.com/Microsoft/clarity-js/pull/139#discussion_r261791739) [](ancestors = 261791739)

Reply:
Done

---
In reply to: [261793020](https://github.com/Microsoft/clarity-js/pull/139#discussion_r261793020) [](ancestors = 261793020,261791739)